File: 530 - Minimum Absolute Difference in BST.java
// import javax.swing.tree.TreeNode;

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    int t = -1;
    int diff = Integer.MAX_VALUE;
    public int getMinimumDifference(TreeNode root) {
        dfs(root);
        return diff;
    }
    public void dfs(TreeNode root) {
        if (root == null) {
            return;
        }
        dfs(root.left);
        if (t == -1) {
            t = root.val;
        } else{
            diff = Math.min(diff, root.val - t);
            t = root.val;
        }
        dfs(root.right);
    }
}

File: 530 - Minimum Absolute Difference in BST.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
let t = -1;
let diff = Number.MAX_SAFE_INTEGER;
var getMinimumDifference = function (root) {
    dfs(root);
    t = -1;
    a = diff;
    diff = Number.MAX_SAFE_INTEGER;
    return a;
};
function dfs(root) {
    if (!root) {
        return;
    }
    dfs(root.left);
    if (t == -1) {
        t = root.val;
    } else {
        diff = Math.min(diff, root.val - t);
        t = root.val;
    }
    dfs(root.right);

}
